### PR TITLE
Assignment4

### PR DESCRIPTION
**Pull request by enrimmartel**

### CODE REVIEW

**Review by dgarijo — CHANGES_REQUESTED** [review]
Please address pending changes

File: Assignment4/enrimmartel-23A254/task06.py [hunk 1]
```diff
@@ -0,0 +1,93 @@
...
+# Visualize the results
+for s, p, o in g:
+  print(s,p,o)
+
+"""**TASK 6.5: Add UPM as the university where John Smith works**"""
+
+# TO DO
+
+g.add((ns.UPM, RDF.type, ns.University))
+g.add((ns.JohnSmith, vcard.WorkFor, ns.UPM))
```
[the remark this answers is not in the document]
Comment (enrimmartel) [reply Assignment4/enrimmartel-23A254/task06.py]: I have looked for a new property. In this case it is hasPost.

File: Assignment4/enrimmartel-23A254/task07.py [hunk 2]
```diff
@@ -0,0 +1,200 @@
...
+"""**TASK 7.1: List all subclasses of "LivingThing" with RDFLib and SPARQL**"""
+
+# TO DO
+from rdflib.plugins.sparql import prepareQuery
+
+# 1) RDFLib
+print("---RDFLib---")
+ns = Namespace("http://somewhere#")
+for s, p, o in g.triples((None, RDFS.subClassOf, ns.LivingThing)):
+    print(s)
```
[the remark this answers is not in the document]
Comment (enrimmartel) [reply Assignment4/enrimmartel-23A254/task07.py:47]: Updated

File: Assignment4/enrimmartel-23A254/task07.py [hunk 3]
```diff
@@ -0,0 +1,200 @@
...
+
+
+# 2) SPARQL
+print("---SPARQL---")
+
+q1 = prepareQuery('''
+    SELECT ?subClass
+    WHERE {
+        ?subClass rdfs:subClassOf ns:LivingThing .
```
[the remark this answers is not in the document]
Comment (enrimmartel) [reply Assignment4/enrimmartel-23A254/task07.py]: Updated

File: Assignment4/enrimmartel-23A254/task07.py [hunk 4]
```diff
@@ -0,0 +1,200 @@
...
+q2a = prepareQuery('''
+    SELECT ?i
+    WHERE {
+        {
+          ?i rdf:type ns:Person.
+        }
+    }
+    ''',
+    initNs={"rdf": RDF, "rdfs": RDFS, "ns": ns}
+)
```
[the remark this answers is not in the document]
Comment (enrimmartel) [reply Assignment4/enrimmartel-23A254/task07.py]: Updated

File: Assignment4/enrimmartel-23A254/task07.py [hunk 5]
```diff
@@ -0,0 +1,200 @@
...
+#Individuals of Person + individuals of subclassesof Person
+q2b = prepareQuery('''
+    SELECT ?i
+    WHERE {
+        {
+          ?i rdf:type ns:Person.
+        }
+        UNION
+        {
+          ?i rdf:type ?subClass.
```
[the remark this answers is not in the document]
Comment (enrimmartel) [reply Assignment4/enrimmartel-23A254/task07.py:104]: Updated

File: Assignment4/enrimmartel-23A254/task07.py [hunk 6]
```diff
@@ -0,0 +1,200 @@
...
+# TO DO
+# Visualize the results
+
+# 1) RDFLib
+print("---RDFLib---")
+ns = Namespace("http://somewhere#")
+
+#Individuals of Person
+for s,p,o in g.triples((None,RDF.type,ns.Person)):
+  print(s)
```
[the remark this answers is not in the document]
Comment (enrimmartel) [reply Assignment4/enrimmartel-23A254/task07.py]: Updated

**Comment by enrimmartel** [comment]
Updated

**Comment by enrimmartel** [comment]
Comments processed :)


---
Enrique

**Review by dgarijo — APPROVED** [review]

**Pull request merged**